src/main/java/org/sefglobal/scholarx/controller/UserController.java

```diff
@@ -0,0 +1,41 @@
+package org.sefglobal.scholarx.controller;
+
+import org.sefglobal.scholarx.exception.ResourceNotFoundException;
+import org.sefglobal.scholarx.model.Mentee;
+import org.sefglobal.scholarx.model.Mentor;
+import org.sefglobal.scholarx.service.MenteeService;
+import org.sefglobal.scholarx.service.MentorService;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+    private final MentorService mentorService;
+    private final MenteeService menteeService;
+
+    public UserController(MentorService mentorService, MenteeService menteeService) {
+        this.mentorService = mentorService;
+        this.menteeService = menteeService;
+    }
+
+    @PostMapping("/mentor")
+    @ResponseStatus(HttpStatus.CREATED)
+    public Mentor applyAsMentor(@RequestParam long programId) throws ResourceNotFoundException {
```

Comment:
Get a `Mentor` object from the request body. 
**Remember `MentorState` should not be in the object as it always should be in `Pending` state.**

Reply:
So, does it send the profile Id with the request body?

Reply:
No only application and prerequisites of mentor.
```
{
  "application": "",
  "prerequisites": ""
}

```
profile id should be in the header but for now keep it hardcoded

Reply:
Right. Thanks